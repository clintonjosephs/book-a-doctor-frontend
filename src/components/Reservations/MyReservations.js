import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import style from './MyReservation.module.css';
import { convertDateWithName } from '../../helpers/format/format';

import getReservations from '../../redux/reservation/reservationThunks';

function MyReservations() {
  const dispatch = useDispatch();
  const reservation = useSelector((state) => state.reservationReducer);
  useEffect(() => {
    if (reservation.length === 0) {
      dispatch(getReservations());
    }
  }, []);

  return (
    <section className={style.reservation}>
      <table className={style.table} id="table">
        <thead>
          <tr>
            <th>Doctor Picture</th>
            <th>Doctor Name</th>
            <th>Doctor Special</th>
            <th>Appointment Date</th>
            <th>action</th>
          </tr>
        </thead>
        <tbody>
          {reservation.length === 0
            ? <tr><td>There is no Reservations</td></tr>
            : reservation.map((element) => (
              <tr key={element.id}>
                <td><img src={element.imageUrl} alt="doctors" className={style.img} /></td>
                <td>{element.doctor.name}</td>
                <td>{element.doctor.specialization}</td>
                <td>{convertDateWithName(element.date_of_appointment)}</td>
                <td><button type="button" className={style.button}>Cancel</button></td>
              </tr>
            ))}
        </tbody>
      </table>
    </section>
  );
}

export default MyReservations;
