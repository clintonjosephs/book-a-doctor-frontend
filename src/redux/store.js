import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import bookDoctorReducer from './bookDoctor/doctorActions';
import userReducer from './user/userActions';
import reservationReducers from './reservation/reservationActions';

const reducer = combineReducers({
  bookDoctorReducer,
  userReducer,
  reservationReducers,
});

// creating a store
const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
