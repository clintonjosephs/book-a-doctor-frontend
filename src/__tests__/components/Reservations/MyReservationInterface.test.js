import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/store';
import MyReservations from '../../../components/Reservations/MyReservations';

const Wrapper = () => (
  <Provider store={store}>
    <Router>
      <MyReservations />
    </Router>
  </Provider>
);

describe('check login components', () => {
  it('matches actual design', () => {
    const component = renderer.create(<Wrapper />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
