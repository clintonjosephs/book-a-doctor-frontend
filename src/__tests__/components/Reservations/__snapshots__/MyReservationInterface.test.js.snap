// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check login components matches actual design 1`] = `
<section
  className="reservation"
>
  <div
    className="table"
    id="table"
  />
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    Count:
    0
  </div>
</section>
`;
